libs/domains/services/feature/src/lib/pods-metrics/pods-metrics.tsx

```diff
@@ -9,60 +28,256 @@ export interface PodsMetricsProps {
 }
 
 export function PodsMetrics({ environmentId, serviceId }: PodsMetricsProps) {
-  const { data = [], isLoading } = useMetrics({ environmentId, serviceId })
+  const {
+    data: metrics = [],
+    isLoading: isMetricsLoading,
+    isError: isMetricsError,
+  } = useMetrics({ environmentId, serviceId })
+  const {
+    data: runningStatuses,
+    isLoading: isRunningStatusesLoading,
+    isError: isRunningStatusesError,
+  } = useRunningStatus({ environmentId, serviceId })
+  const [sorting, setSorting] = useState<SortingState>([{ id: 'state', desc: false }])
+  const {
+    data: service,
+    isLoading: isServiceLoading,
+    isError: isServiceError,
+  } = useService({ environmentId, serviceId })
 
-  if (data === null) {
-    return null
+  const pods: Pod[] = useMemo(() => {
+    // NOTE: metrics or runningStatuses could be undefined because backend doesn't have the info.
+    // So we must find all possible pods by merging the two into a Set.
+    const podNames = new Set([
+      ...(runningStatuses?.pods.map(({ name }) => name) ?? []),
+      ...(metrics?.map(({ pod_name }) => pod_name) ?? []),
+    ])
+
+    return [...podNames].map((podName) => ({
+      ...(metrics?.find(({ pod_name }) => pod_name === podName) ?? {}),
+      ...(runningStatuses?.pods.find(({ name }) => name === podName) ?? {}),
+      podName,
+    }))
+  }, [metrics, runningStatuses])
+
+  const columnHelper = createColumnHelper<Pod>()
+  const placeholder = <Icon name={IconAwesomeEnum.CIRCLE_QUESTION} className="text-neutral-300" />
+
+  const containerImage = match(service)
+    .with({ serviceType: ServiceTypeEnum.JOB }, ({ source }) => source?.image)
+    .with({ serviceType: ServiceTypeEnum.CONTAINER }, ({ image_name, tag, registry }) => ({
+      image_name,
+      tag,
+      registry,
+    }))
+    .otherwise(() => undefined)
+
+  const columns = useMemo(
+    () => [
+      columnHelper.accessor('podName', {
+        header: match(service)
+          .with({ serviceType: ServiceTypeEnum.JOB }, () => 'Job executions')
+          .otherwise(() => 'Service pods'),
+        cell: (info) => {
+          const podName = info.getValue()
+          return (
+            <div className="flex flex-row">
+              <button
+                className="w-9 pointer text-neutral-350"
+                type="button"
+                onClick={info.row.getToggleExpandedHandler()}
+              >
+                {info.row.getIsExpanded() ? (
+                  <Icon name={IconAwesomeEnum.CHEVRON_UP} />
+                ) : (
+                  <Icon name={IconAwesomeEnum.CHEVRON_DOWN} />
```

Comment:
This is not so easy because tanstack-table tend to re-render the whole table.
In addition, Audit logs doesn't have transition so I guess we could tackle that later.